src/Angor/Client/Pages/Signatures.razor

```diff
@@ -161,6 +196,50 @@
         }
     }
 
+    private async Task RefreshSignaturesInternal()
+    {
+        if (passwordComponent.HasPassword())
+        {
+            await RefreshSignatures();
+        }
+        else
+        {
+            passwordComponent.ShowPassword(async () =>
+            {
+                await RefreshSignatures();
+            });
+        }
+    }
+
+    private async Task RefreshSignatures()
+    {
+        if (isCooldown)
+            return;
+        
+        refreshButtonSpinner = true;
```

Comment:
`refreshButtonSpinner` and `isCooldown` can be combined to one variable (that worked in other paces I think it will work here too)